src/subscript-loc.c

```diff
@@ -142,27 +142,32 @@ r_obj* int_as_location(r_obj* subscript,
       if (opts->missing == SUBSCRIPT_MISSING_ERROR) {
         stop_subscript_missing(subscript, opts);
       }
-    } else {
-      if (elt < 0) {
-        switch (opts->loc_negative) {
-        case LOC_NEGATIVE_INVERT: return int_invert_location(subscript, n, opts);
-        case LOC_NEGATIVE_ERROR: stop_location_negative(subscript, opts);
-        case LOC_NEGATIVE_IGNORE: break;
-        }
+    } else if (elt == 0) {
+      switch (opts->loc_zero) {
+      case LOC_ZERO_REMOVE: ++n_zero; break;
+      case LOC_ZERO_ERROR: stop_location_zero(subscript, opts);
+      case LOC_ZERO_IGNORE: break;
       }
-
-      if (elt == 0) {
-        switch (opts->loc_zero) {
-        case LOC_ZERO_REMOVE: ++n_zero; break;
-        case LOC_ZERO_ERROR: stop_location_zero(subscript, opts);
-        case LOC_ZERO_IGNORE: break;
-        }
-      } else if (abs(elt) > n) {
-        switch (opts->loc_oob) {
-        case LOC_OOB_ERROR: stop_subscript_oob_location(subscript, n, opts);
-        case LOC_OOB_EXTEND:
-        case LOC_OOB_REMOVE: ++n_oob; break;
+    } else if (elt < 0) {
+      switch (opts->loc_negative) {
+      case LOC_NEGATIVE_INVERT: return int_invert_location(subscript, n, opts);
+      case LOC_NEGATIVE_ERROR: stop_location_negative(subscript, opts);
+      case LOC_NEGATIVE_IGNORE: {
+        if (abs(elt) > n) {
+          switch (opts->loc_oob) {
+          case LOC_OOB_ERROR: stop_subscript_oob_location(subscript, n, opts);
+          case LOC_OOB_EXTEND: stop_subscript_negative_oob_location(subscript, n, opts);
```

Comment:
These purposefully use different error messages

With `oob = "error", negative = "ignore"`, we want to report both negative and positive oob locations

With `oob = "extend", negative = "ignore"`, we only want to report negative oob locations (because positive oob locations might not actually be oob, they might be valid `"extend"` indices)

I've added tests for both of these cases